apis/v1alpha2/validation/gateway.go

```diff
@@ -44,7 +45,32 @@ func validateGatewaySpec(spec *gatewayv1a2.GatewaySpec, path *field.Path) field.
 // to the Gateway API specification.
 func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	// TODO [danehans]: Add additional validation of listener fields.
-	return validateListenerHostname(listeners, path)
+	var errs field.ErrorList
```

Comment:
nit: 
```go
var errs field.ErrorList
errs = append(errs, validateListenerTLSConfig(listeners, path)...)
errs = append(errs, validateListenerHostname(listeners, path)...)
return errs
```

otherwise we will end up with e3, e4, ..., e99

Reply:
seems fair to me will fix this :) . Thanks for your review

Reply:
Fixed in the latest patch